### PR TITLE
Json

### PR DESCRIPTION
**Pull request by Bairdo**
Changed urlencoding of HTTP methods to json, this will allow the usernames to contain special characters.

### CODE REVIEW

**Review by libunamari — APPROVED** [review]

File: capflow/CapFlow.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ def install_dns_fwd(nw_src, nw_dst, out_port, src_port):
             self._contr.add_flow(datapath,
                                  2001,
                                  parser.OFPMatch(
-                                     eth_src=nw_dst,
+                                     #eth_src=nw_dst,
```
Comment (Bairdo) [inline capflow/CapFlow.py]: remove this line, if its not needed

File: capflow/CapFlow.py [hunk 2]
```diff
@@ -250,17 +250,17 @@ def install_dns_fwd(nw_src, nw_dst, out_port, src_port):
                                      [parser.OFPActionOutput(out_port)])],
                                  0,
                                  self._table_id_cf,
-                                 msg=msg, in_port=in_port, idle_timeout=30
+                                 msg=msg, in_port=in_port, idle_timeout=30, packet_out=False
                                 )
 	    # is this actually needed.
-            out = parser.OFPPacketOut(
+            '''out = parser.OFPPacketOut(
```
Comment (Bairdo) [inline capflow/CapFlow.py]: again if this not needed remove it.

File: capflow/CapFlow.py [hunk 3]
```diff
@@ -414,6 +414,7 @@ def deal_with_arp(self, datapath, msg, pkt, eth, in_port, nw_dst, nw_src):
                 buffer_id=msg.buffer_id,
                 in_port=in_port,
                 actions=[datapath.ofproto_parser.OFPActionOutput(port)],
+                #actions=[datapath.ofproto_parser.OFPInstructionGotoTable(4)], #TODO
```
Comment (Bairdo) [inline capflow/CapFlow.py:407]: REMOVE again

File: capflow/rest.py [hunk 4]
```diff
@@ -75,42 +71,37 @@ def validate(address):
...
-    def delete(self, req, ip, user, **_kwargs):
         if self.capflow_interface.is_authed(ip) is False:
-            print "user already logged off"
+            print "user not logged in"
             return Response(status=200)
 
         print "HTTP Delete : " + user + " " + ip
-	# todo remove the flows somehow.
-
+        print authJSON
```
Comment (Bairdo) [inline capflow/rest.py]: remove or use correct logging

File: capflow/rest.py [hunk 5]
```diff
@@ -75,42 +71,37 @@ def validate(address):
...
 	self.capflow_interface.log_client_off(ip, user)
 
         return Response(status=200)
 
+    def authenticate_post(self, req, **kwargs):
+        try:
+            authJSON = json.loads(req.body)
+        except:
+            return Response(status=400, body="Unable to parse JSON")
+        print authJSON
```
Comment (Bairdo) [inline capflow/rest.py]: remove or use correct logging, again

**Review by Bairdo — COMMENTED** [review]
remove the stuff that is commented out, and not needed. If they 'might' be needed they are in the git history anyway.

**Pull request merged**